File: python/examples/get_filing_products_offerings.py
import pprint
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from request import BaseRequest
from dotenv import dotenv_values

config = dotenv_values()

# Example of GET /filing-products/offerings

class GetFilingProductsOfferingsRequest(BaseRequest):
    def __init__(self):
        super().__init__()
    
    def get_filing_products_offerings(self, company_id, jurisdiction):
        params = {
            'company_id': company_id,
            'jurisdiction': jurisdiction
        }
        return self.make_request('GET', '/filing-products/offerings', params=params)

# run as standalone script by passing any command line argument
if len(sys.argv) > 1:
    company_id = config['COMPANY_ID']
    jurisdiction = config['JURISDICTION']

    request = GetFilingProductsOfferingsRequest()
    response = request.get_filing_products_offerings(company_id, jurisdiction)

    pprint.pprint(response)

File: javascript/examples/get_filing_products_offerings.js
const baseRequest = require('../base_request.js').baseRequest

const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

// Example of GET /filing-products/offerings

const COMPANY_ID	= process.env.COMPANY_ID;
const JURISDICTION 	= process.env.JURISDICTION;

params = {
	company_id: COMPANY_ID,
  	jurisdiction: JURISDICTION
}

token = baseRequest.request.token({ path: '/filing-products/offerings' });

baseRequest.request.get({ path: '/filing-products/offerings', token: token, queryParams: params });